src/Application/Components/MapCanvas/MapCanvas.jsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { fabric } from 'fabric-webpack';
+
+/**
+ * No Pin functionality yet.
+ */
+const MapCanvas = React.createClass({
+  propTypes: {
+    pins: React.PropTypes.object,
```

Comment:
There's a number of ways, but the basic notion is get prop / state to a place where `===` reliably 100% tells you if it's different.  Objects are mutable, and `===` tests by memory location for objects, not any sort of deep comparison, so somewhere above this component someone could say `pin.x = 120`, thus mutating the pin, but we have no way of knowing unless we do a deep compare.  There are a number of ways having immutable objects is beneficial, but the biggest one here is `PureRenderMixin` won't work without it.  If you make a class with only a constructor and getters (we ignore for the moment that javascript is the wild west and someone could go into a class instance and change whatever they want) then it's effectively immutable.  The other choice for avoiding mutable objects is using `Immutable.Map`.  The choice for avoiding mutable arrays is using `Immutable.List`.  So `pins` would become 
```
ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
  xPosition: React.PropTypes.number.isRequired,
  yPosition: React.PropTypes.number.isRequired,
})
```
Or, if you make Pin a class (altho, if it is just x and y a more descriptive name might be `OrderedPair`), then you could have `Immutable.listOf(React.PropTypes.instanceOf(OrderedPair)` which I kinda like better, plus I <3 classes